gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -40,43 +54,71 @@ class TestCaseFailure(Exception):
 class TestRunner(object):
   """Runs the tests using pipelines API."""
 
-  def __init__(self, tests):
-    # type: (List[Union[PreprocessorTestCase, VcfToBQTestCase]]) -> None
+  def __init__(self, tests, revalidation=None):
+    # type: (List[List[TestCaseInterface]], str) -> None
+    """Initializes the TestRunner.
+
+    Args:
+      tests: All test cases.
+      revalidation: If not none, skip most of the test (including table creation
+        and populating for vcf_to_bq) and only do the result validation.
+    """
     self._tests = tests
     self._service = discovery.build(
         'genomics',
         'v1alpha2',
         credentials=GoogleCredentials.get_application_default())
-    self._operation_names = []  # type: List[str]
-    self._responses = []  # List[Dict]
+    self._revalidation = revalidation
+    self._operation_names_to_test_states = {}  # type: Dict[str, TestCaseState]
 
   def run(self):
     """Runs all tests."""
-    for test in self._tests:
-      # The following pylint hint is needed because `pipelines` is a method that
-      # is dynamically added to the returned `service` object above. See
-      # `googleapiclient.discovery.Resource._set_service_methods`.
-      # pylint: disable=no-member
-      request = self._service.pipelines().run(body=test.pipeline_api_request)
-      operation_name = request.execute()['name']
-      self._operation_names.append(operation_name)
+    if self._revalidation:
+      for test_cases in self._tests:
+        for test_case in test_cases:
+          test_case.validate_result()
+          return
```

Comment:
this `return` should be on the `if` level? I also think that it's more readable in this case to just do an` else` with the rest of the logic since the `else` part isn't that large.